### DISCUSSION TITLE
Controls

### DISCUSSION THREAD

**Discussion by fredchristianson** [fredchristianson/javascript-game-engine#11] [Ideas]
@chylinski82 brought up controls in the requirements discussion and it needs its own discussion.  What types of controls do we want the core to handle
- keyboard:  Are "directions" a special combo (up/down/left/right or a/w/s/z could be a direction pad)
- mouse: buttons. movements. speed.
- touch: single and multi touch.  location, movement, & speed?
- combinations:  ctrl+mouse button

I think games should be able to specify options.  Different sets of controls for keyboard, mouse, or touch depending on users choice.  The app would manage the settings.  If game has generic consoles (direction pad) the app can let the user pick which keys are used.

What other specific (ctrl-a) and generic (direction pad) controls should we look at?  A game can always handle everything it wants, but providing common implementations will make games easier to write.  We also want to define interactions between controls and game entities.

**Comment by chylinski82** [comment]
since, we're talking about controls it borught to my mind somethin even more important perhaps it should be clarify in a top level discussion (is there a top level discussion?), what is the main device that this project will target? the styles will depend on it and lot of other things that i probably don't even have an idea about. I think in 2023 it should be touch screen/phone, but I'm open to keyboard/ desktop as the main device. 

as per specific questions:

- keyboard: I would prefer arrows, but 'a','w','s' and 'd' are fine too. some games may need different layout, this is a game i made in Phaser.js `http://android-coop.surge.sh` and keyboard controls are  'q', 'a', 'p' and 'l', the standard way just didn't make sense.

- mouse: really depend on a game, you may need buttons for different actions, some games where speed doesn't matter you could just left click where the Actor should go or even left click on a set of arrows in a corner of the screen, which would be there for touchscreen anyway

- touch: I didn't even know there was multi touch, i guess how i forget how much you can do with a phone, i just had in mind a set of arrows/button to tap on or something like that
